pkg/forwarder/worker.go

```diff
@@ -85,7 +89,14 @@ func (w *Worker) Start() {
 		defer close(w.stopped)
 
 		for {
-			// handling high priority transactions first
+			// poll for connection reset events first
+			select {
```

Comment:
This works, and it would respect the interval for the connection resets in a time-sensitive fashion something the alternative I'm about to suggest wouldn't do as well. But I was also wondering if it didn't make sense to trigger the connection reset in `callProcess` moving the select block in there and keeping these selects a little cleaner. No strong opinion, just a thought.

Reply:
Yes, feels like better design actually, the "reset" itself should be almost instant (I don't know if we should be concerned about `Client.CloseIdleConnections` potentially blocking for a while though, I'll check). I'll address

Reply:
should be addressed now, thanks for the suggestion!

(and I don't think we need to handle run `w.Client.CloseIdleConnections` asynchronously, this shouldn't block for too long)